state/postgres_database.go

```diff
@@ -21,4 +21,33 @@ type PostgresDatabase struct {
 	// This is used to track whether the database needs to be vacuumed in order to prevent multixact ID wraparound or to
 	// allow pg_multixact to be shrunk. It is the minimum of the per-table pg_class.relminmxid values.
 	MinimumMultixactXID Xid
+
+	FrozenXIDAge uint32 // Age of FrozenXID
+	MinMXIDAge   uint32 // Age of MinimumMultixactXID
+
+	XactCommit   int64 // Number of transactions in this database that have been committed
+	XactRollback int64 // Number of transactions in this database that have been rolled back
+}
+
+// PostgresDatabaseStats - Database statistics for a single database
+type PostgresDatabaseStats struct {
+	TransactionCount uint64 // Transaction count of this database
+}
+
+// PostgresDatabaseStatsMap - Map of database statistics (key = database Oid)
+type PostgresDatabaseStatsMap map[Oid]PostgresDatabaseStats
+
+// DiffedPostgresDatabaseStat - Database statistics for a single database as a diff
+type DiffedPostgresDatabaseStats struct {
+	TransactionsPerSecond uint64
+}
+
+// DiffedDatabaseStats - Map of diffed database statistics (key = database Oid)
+type DiffedPostgresDatabaseStatsMap map[Oid]DiffedPostgresDatabaseStats
+
+// DiffSince - Calculate the diff between two stats runs
+func (curr PostgresDatabaseStats) DiffSince(prev PostgresDatabaseStats, collectedIntervalSecs uint32) DiffedPostgresDatabaseStats {
+	return DiffedPostgresDatabaseStats{
+		TransactionsPerSecond: (curr.TransactionCount - prev.TransactionCount) / uint64(collectedIntervalSecs),
```

Comment:
Whilst we are using this pattern for the system statistics (to divide by `collectedIntervalSecs` in the collector), we don't do this for any other statistics - typically we've done this division on the application side, and just passed along the difference between the counter values. For system statistics its because some providers already give us a "per second" value, so we normalize all providers to work that way.

I have a preference to not the division here (i.e. not divide by collectedIntervalSecs as this is doing right now), because its sometimes helpful for debugging to see the difference value that a snapshot calculated - vs in this case its likely there is some rounding due to this devision, making debugging harder.

(e.g. when a counter wraps around, there are sometimes details that can be understood from the actual diff value between two full snapshot runs)